torch_xla_py/xla_dist.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+"""Tool to distribute training on Cloud TPU Pods."""
+from __future__ import division
+from __future__ import print_function
+
+
+class Worker(object):
```

Comment:
I'd structure different. Given, today, we still have a dual instance (Client/Service), I'd have:
```Python
class Worker(object):
  # All the common worker fields/methods.

class ClientWorker(Worker):
  # Client worker specific bits. It's OK is none ATM.

class ServiceWorker(Worker):
  # Service worker specific bits. It's OK is none ATM.

class Cluster(object):
  # This is a complete description of a Cluster, in terms of Client and Service workers.
  def __init__(self, client_workers, service_workers):
    pass
```

Reply:
So I had done similar in this [commit](https://github.com/pytorch/xla/pull/864/commits/a075c1c0653d2c68c6c8ee3edcb6534705391353) (except having separate `Cluster` objects). But I didn't see any real bits that are client vs service specific and that being the case it felts like having that distinction when they're just the same thing seemed to bloat stuff. What do you think?

Also if we do separate Client vs Service for worker side, I think we should separate the clusters too since their `validate()` method would be worker specific in that case.

Reply:
> So I had done similar in this [commit](https://github.com/pytorch/xla/pull/864/commits/a075c1c0653d2c68c6c8ee3edcb6534705391353) (except having separate `Cluster` objects). But I didn't see any real bits that are client vs service specific and that being the case it felts like having that distinction when they're just the same thing seemed to bloat stuff. What do you think?

IMHO, since we are still in 2VM, I'd keep the distinction.
All the code will go in `Worker` anyway, but having the distinction leads to a more precise object model.

> Also if we do separate Client vs Service for worker side, I think we should separate the clusters too since their `validate()` method would be worker specific in that case.

The `validate()` of the `Cluster` can still do the proper job since it holds the two sets, no?

Reply:
Oh you're saying we hold both client and service workers in a single set. Yeah sure. We can do that. We might want to change the `ClusterResolver` apis then? Such that we only have a single method that returns a `Cluster` with both client and service meshes, instead of `vm_cluster()` and `tpu_cluster()` methods separately. Unless you want to return partial `Cluster`s from each `ClusterResolver` method. IMHO, I don't think the `ClientWorker` and `ServiceWorker` vs `Worker` matters that much at this point.

Reply:
Not in a single set.
The Cluster holds both sets.
Yes, change CusterResolver. A Cluster should describe the whole thing.
Otherwise we'd need to have a ClientCluster and ServiceCluster.
Which IMHO does not seem a better approach.

Reply:
Done.